### PR TITLE
Divanovic/llama tg demo

### PR DESCRIPTION
**Pull request by djordje-tt**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/13102
https://github.com/tenstorrent/tt-metal/issues/11730
### Problem description
Enable prefill+decode demo

### What's changed
- Added support for prefill+decode demo
- Removed model_config dependency to t3k model_config and created TG one
- Separated all component configs from it's files to model_config and made components cleaner and more readable

### Checklist
- [ ] https://github.com/tenstorrent/tt-metal/actions/runs/11034614348

### CODE REVIEW

File: models/demos/t3000/llama2_70b/tt/llama_common.py [hunk 1]
```diff
@@ -174,11 +175,18 @@ def setup_llama_env(llama_version="llama3", max_batch_size=32, max_context_len=4
     logger.info(f"Tokenizer file: {tokenizer_path}")
     logger.info(f"Cache directory: {cache_path}")
 
-    model_config = get_model_config(
-        llama_version=llama_version,
-        max_batch_size=max_batch_size,
-        max_context_len=max_context_len,
-    )
+    if llama_version == "llama3-tg":
+        model_config = get_galaxy_model_config(
```
Comment (cglagovichTT) [inline models/demos/t3000/llama2_70b/tt/llama_common.py:179]: What does Llama TG use get_model_config for? Can we get rid of it?

File: models/demos/tg/llama3_70b/tt/llama_model_galaxy.py [hunk 2]
```diff
@@ -256,9 +227,9 @@ def prepare_inputs(self, inp_ids, start_pos, valid_seq_len=None, attn_mask=None,
             )
 
             attn_masks = None
-        elif mode == "prefill":
-            assert seq_len % 128 == 0 and seq_len > 0, "Prefill mode only supports seq_len > 0 and seq_len % 128"
 
+        elif mode == "prefill":
+            assert seq_len % 32 == 0 and seq_len > 0, "Prefill mode only supports seq_len > 0 and seq_len % 32"
```
[the remark this answers is not in the document]
Comment (djordje-tt) [reply models/demos/tg/llama3_70b/tt/llama_model_galaxy.py:232]: Colman mentioned that ROPE is not working with something not power of 32.

[the remark this answers is not in the document]
Comment (cglagovichTT) [reply models/demos/tg/llama3_70b/tt/llama_model_galaxy.py:232]: See how Llama t3k handles input validation https://github.com/tenstorrent/tt-metal/blob/012a1671b5a6a3c98a07ffadd62eec0b0b605523/models/demos/t3000/llama2_70b/tt/llama_model_optimized.py#L161

[the remark this answers is not in the document]
Comment (cglagovichTT) [reply models/demos/tg/llama3_70b/tt/llama_model_galaxy.py:232]: Rope requires a power of 2 afaik

[the remark this answers is not in the document]
Comment (djordje-tt) [reply models/demos/tg/llama3_70b/tt/llama_model_galaxy.py:232]: `validate_input_shape` only checks whether `seq_len <= max_context_len`, I didn't see anything related to the power of 2.

**Comment by djordje-tt** [comment]
CI frequent test run: https://github.com/tenstorrent/tt-metal/actions/runs/11034614348

**Review by cglagovichTT — APPROVED** [review]

**Pull request merged**